### PR TITLE
Changing over to just install the recommender, CRD, and RBAC. Add an upgrade path

### PR DESCRIPTION
**Pull request by sudermanjr**

### CODE REVIEW

**Comment by rbren** [comment]
Why no more `upgradeVPA` flag? Guessing that'd just be identical to running `installVPA` a second time?

Can we call it something other than `uninstallVPA`? Maybe `reinstallVPA`, or `uninstallOldVPA`?

My plan is to add those flags to the default insights-agent install command for folks running appropriate versions of the chart - it'd look a little broken to have both `--set installVPA=true --set uninstallVPA=true`

**Comment by sudermanjr** [comment]
> Why no more `upgradeVPA` flag? Guessing that'd just be identical to running `installVPA` a second time?
> 
> Can we call it something other than `uninstallVPA`? Maybe `reinstallVPA`, or `uninstallOldVPA`?
> 
> My plan is to add those flags to the default insights-agent install command for folks running appropriate versions of the chart - it'd look a little broken to have both `--set installVPA=true --set uninstallVPA=true`

Yeah, so the upgrade is now identical due to the `kubectl apply` being in there. That's why it was removed.

We _could_ call it reinstallVPA, but it doesn't really do that, it just uninstalls.  I suppose we _could_ just make it uninstall and install in a single go, but I wanted the option to make it a post-delete hook as well just in case. 

How about `preUninstallVPA` or something like that?

**Comment by sudermanjr** [comment]
I added a `reinstallVPA` flag that combines the two in order to make the command more palatable

File: stable/goldilocks/templates/vpa-uninstall-hook.yaml [hunk 1]
```diff
@@ -1,65 +1,65 @@
-{{- if .Values.upgradeVPA }}
+{{- if or (eq .Values.uninstallVPA true) (eq .Values.reinstallVPA true) }}
```
Comment (rbren) [inline stable/goldilocks/templates/vpa-uninstall-hook.yaml]: Maybe simplify to `if or .Values.uninstallVPA .Values.resinstallVPA`

**Review by rbren — APPROVED** [review]

File: stable/goldilocks/templates/vpa-uninstall-hook.yaml (hunk 1 above)
Comment (sudermanjr) [reply stable/goldilocks/templates/vpa-uninstall-hook.yaml]: Oh sure, why not.  I was being lazy and copying from somewhere ;-)

**Pull request merged**